forge-gui/res/cardsfolder/upcoming/corruption_of_towashi.txt

```diff
@@ -0,0 +1,14 @@
+Name:Corruption of Towashi
+ManaCost:4 U
+Types:Enchantment
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigIncubate | TriggerDescription$ When CARDNAME enters the battlefield, incubate 4. (Create an Incubator token with four +1/+1 counters on it and "{2}: Transform this artifact." It transforms into a 0/0 Phyrexian artifact creature.)
+SVar:TrigIncubate:DB$ Incubate | Amount$ 4
+T:Mode$ Transformed | ValidCard$ Permanent.YouCtrl | Execute$ TrigDraw | TriggerZones$ Battlefield | CheckSVar$ X | SVarCompare$ LT1 | OptionalDecider$ You | TriggerDescription$ Whenever a permanent you control transforms or a permanent enters the battlefield under your control transformed, you may draw a card. Do this only once each turn.
```

Comment:
StoreSVar shouldn't be needed, we have `ResolvedLimit$ 1` for it

Reply:
Ah, since the overriding ability is the same for both triggers – NICE

Reply:
I'm not 100% sure this is fool proof

I mean yeah it's the same SVar, but when the trigger are built, isn't it creating two SpellAbility objects out of it?

Reply:
Ah good point, now I'm not so sure
It might be that even if the script uses the same SVar the triggers will not share it :/

Reply:
I guess we can't do it

We could teach CardFactory to reuse the SA - but some other cards will have two triggers that are separate abilities which might still work with the same SVar

So it would require manual tagging and that seems too fiddly imo

Sorry about that

Reply:
Hey no worries – I actually meant to ask about this as I wasn't sure the SVars were good enough

Maybe there's some way to  link triggers of different trigger types for this sort of thing – not this PR though

Reply:
It did seem to be working in testing... I was doing stuff with the new Elesh Norn so lots of opportunities for both of the triggers to go off.

Reply:
You're right it works

Need to debug why :D

Reply:
The problem is the Transformed property fails because for some reason the card is still in OriginalState

So your ChangesZone never triggers but your Transformed actually does for Elesh Norn too because of missing `+inZoneBattlefield` most likely

Reply:
Oops, I see, it's all fixing Corruption (as you know)